### PR TITLE
Get Default Branch of each repo

### PR DESCRIPTION
**Pull request by guineveresaenger**
Fixes https://github.com/pulumi/github-issue-automation/issues/17

### CODE REVIEW

File: lambda/main.go [hunk 1]
```diff
@@ -127,17 +128,24 @@ func LambdaHandler(event NewRelease) error {
 	triggerWorkflowAllowList := strings.Split(os.Getenv("TRIGGER_WORKFLOW_ALLOW_LIST"), " ")
 	if shouldTriggerWorkflow(pulumiRepo, triggerWorkflowAllowList) {
 		log.Printf("Pulumi repo '%s' was found in the allow list. Triggering workflow.", pulumiRepo)
-
+		repoObj, resp, err := gitHubClient.Repositories.Get(context.Background(), "pulumi", pulumiRepo)
+		if resp.StatusCode != http.StatusOK {
+			log.Printf("Pulumi repo %s could not be found: %s", pulumiRepo, resp.Status)
```
Comment (jkodroff) [inline lambda/main.go:133]: This should be `return fmt.Errorf()` - basically a panic.

File: lambda/main.go [hunk 2]
```diff
@@ -127,17 +128,24 @@ func LambdaHandler(event NewRelease) error {
...
+		}
+		if err != nil {
+			return err
+		}
+		ref := repoObj.GetDefaultBranch()
```
Comment (jkodroff) [inline lambda/main.go:138]: I believe this will panic because `repoObj` will be nil if we don't quit above.

File: lambda/main.go (hunk 1 above)
Comment (jkodroff) [reply lambda/main.go:133]: Nit: Also, I believe you may need to accommodate _any_ 2XX response from the API - docs would say for sure.

**Review by jkodroff — CHANGES_REQUESTED** [review]

File: lambda/main.go (hunk 1 above)
Comment (guineveresaenger) [reply lambda/main.go:133]: I'm gonna go with the trust-and-yell-if-it-changes model on this one:
https://docs.github.com/en/rest/reference/repos#get-a-repository <-- returns a 200 OK

**Review by jkodroff — APPROVED** [review]

**Pull request merged**